src/common/storage/project/cache.js

```diff
@@ -250,6 +254,32 @@ define(['common/util/assert', 'common/storage/constants'], function (ASSERT, CON
                 stackedObjects[key] = obj;
             }
         };
+
+        this.insertPatchObject = function (obj) {
+            ASSERT(typeof obj === 'object' && obj !== null);
+
+            var base,
+                patch,
+                key = obj[CONSTANTS.MONGO_ID];
+
+            if (obj.base && typeof obj.patch === 'object' && key) {
+                base = getFromCache(obj.base);
+
+                if (base) {
+                    patch = jsonPatcher.apply(base, obj.patch);
+                    if (patch.status === 'success') {
+                        patch.result[CONSTANTS.MONGO_ID] = key;
+                        this.insertObject(patch.result);
+                    } else {
+                        logger.error('patch application failed', patch);
```

Comment:
I added the patch object to the patch result so it can be easily 'used' in other places as well.